examples/text-classification/run_pl_glue.py

```diff
@@ -1,115 +1,205 @@
 
-        outputs = self(**inputs)
+        outputs = self(**batch)
         loss = outputs[0]
 
         tensorboard_logs = {"loss": loss, "rate": self.lr_scheduler.get_last_lr()[-1]}
         return {"loss": loss, "log": tensorboard_logs}
 
     def prepare_data(self):
-        "Called to initialize data. Use the call to construct features"
-        args = self.hparams
-        processor = processors[args.task]()
-        self.labels = processor.get_labels()
-
-        for mode in ["train", "dev"]:
-            cached_features_file = self._feature_file(mode)
-            if os.path.exists(cached_features_file) and not args.overwrite_cache:
-                logger.info("Loading features from cached file %s", cached_features_file)
-                features = torch.load(cached_features_file)
-            else:
-                logger.info("Creating features from dataset file at %s", args.data_dir)
-                examples = (
-                    processor.get_dev_examples(args.data_dir)
-                    if mode == "dev"
-                    else processor.get_train_examples(args.data_dir)
+        """
+        Use this to download and prepare data.
+        In distributed (GPU, TPU), this will only be called once.
+        This is called before requesting the dataloaders.
+        """
+        self.dataset = nlp.load_dataset("glue", name=self.hparams.task)
+        self.metric = nlp.load_metric("glue", name=self.hparams.task)
+
+        cached_dataset_file = self._feature_file(self.hparams.task)
+
+        if os.path.exists(cached_dataset_file) and not self.hparams.overwrite_cache:
+            logger.info("Loading from cached dataloader file ", cached_dataset_file)
+            self.dataset = torch.load(cached_dataset_file)
+        else:
+            logger.info("No cache file found. Creating one at", cached_dataset_file)
+
+            # Make the directory to hold cached dataset if it doesn't already exist
+            os.makedirs(os.path.dirname(cached_dataset_file), exist_ok=True)
+
+            # We don't know names of text field(s) so we find that here. If multiple, we tokenize text pairs.
+            text_fields = [field.name for field in self.dataset["train"].schema if pyarrow.types.is_string(field.type)]
+
+            def convert_to_features(example_batch):
+                """Function to be mapped across GLUE datasets. Tokenizes text(s) and renames label column.
+
+                Args:
+                    example_batch (Dict[List]): Dict with feature names as keys and batch values as values
+
+                Returns:
+                    Dict[List]: The desired updates to the batch (keys are feature names, values are batch values)
+                """
+
+                # Either encode single sentence or sentence pairs
+                if len(text_fields) > 1:
+                    texts_or_text_pairs = list(zip(example_batch[text_fields[0]], example_batch[text_fields[1]]))
```

Comment:
is call to `list` needed?

Reply:
I'm actually not sure...will investigate. Nice catch!!

Reply:
confirmed not needed